src/main/kotlin/com/refinedmods/refinedstorage/gui/screen/AmountSpecifyingScreen.kt

```diff
@@ -1,45 +1,54 @@
-package com.refinedmods.refinedstorage.screen
+package com.refinedmods.refinedstorage.gui.screen
 
 import com.refinedmods.refinedstorage.RS
 import com.refinedmods.refinedstorage.render.RenderSettings
+import kotlin.math.min
 import net.minecraft.client.MinecraftClient
 import net.minecraft.client.gui.widget.ButtonWidget
 import net.minecraft.client.gui.widget.TextFieldWidget
 import net.minecraft.client.util.math.MatrixStack
-import net.minecraft.entity.player.PlayerInventory
+import net.minecraft.entity.player.PlayerEntity
 import net.minecraft.screen.ScreenHandler
 import net.minecraft.text.Text
 import net.minecraft.text.TranslatableText
 import org.apache.commons.lang3.tuple.Pair
 import org.lwjgl.glfw.GLFW
 
 abstract class AmountSpecifyingScreen<T : ScreenHandler>(
-        val parent: BaseScreen<T>, container: T,
-        width: Int,
-        height: Int,
-        playerInventory: PlayerInventory,
-        title: Text
-) : BaseScreen<T>(container, width, height, playerInventory, title) {
-    protected var amountField: TextFieldWidget? = null
-    protected var okButton: ButtonWidget? = null
-    protected var cancelButton: ButtonWidget? = null
+    val parent: BaseScreen<T>,
+    container: T,
+    private val xSize: Int,
+    private val ySize: Int,
+    player: PlayerEntity,
+    title: Text
+) : BaseScreen<T>(container, player, title) {
+    protected lateinit var amountField: TextFieldWidget
+    protected lateinit var okButton: ButtonWidget
+    protected lateinit var cancelButton: ButtonWidget
     protected abstract val okButtonText: Text
     protected abstract val texture: String
     protected abstract val increments: IntArray
     protected abstract val defaultAmount: Int
     protected abstract fun canAmountGoNegative(): Boolean
     protected abstract val maxAmount: Int
-    protected open val amountPos: Pair<Int, Int>
-        protected get() = Pair.of(7 + 2, 50 + 1)
-    protected open val okCancelPos: Pair<Int?, Int?>?
-        protected get() = Pair.of(114, 33)
-    protected open val okCancelButtonWidth: Int
-        protected get() = 50
+    protected open val amountPos: Pair<Int, Int> = Pair.of(7 + 2, 50 + 1)
```

Comment:
Because the original code did. I only update this screen the bare minimum to get `BaseScreen` to compile. It's not currently used. Could use some TLC at some point. Is there something better than a `Pair` in your opinion?

Reply:
Kotlin `Pair`?

Reply:
Oh, duh. My bad.